ssh-key/src/sshsig.rs

```diff
@@ -346,3 +350,38 @@ impl Encode for SignedData<'_> {
         Ok(())
     }
 }
+
+#[cfg(feature = "serde")]
+impl<'de> Deserialize<'de> for SshSig {
+    fn deserialize<D>(deserializer: D) -> core::result::Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        if deserializer.is_human_readable() {
+            let string = String::deserialize(deserializer)?;
+            string.parse::<SshSig>().map_err(de::Error::custom)
+        } else {
+            let bytes = Vec::<u8>::deserialize(deserializer)?;
+            let mut bytes_slice: &[u8] = &bytes;
+            Self::decode(&mut bytes_slice).map_err(de::Error::custom)
```

Comment:
You should be able to do something like:

```suggestion
            Self::decode(&mut bytes.as_slice()).map_err(de::Error::custom)
```